gcp/ignore_error_predicate.go

```diff
@@ -23,12 +24,21 @@ func isIgnorableError(notFoundErrors []string) plugin.ErrorPredicate {
 // shouldIgnoreErrorPluginDefault:: Plugin level default function to ignore a set errors for hydrate functions based on "ignore_error_codes" config argument
 func shouldIgnoreErrorPluginDefault() plugin.ErrorPredicateWithContext {
 	return func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, err error) bool {
-		if !hasIgnoredErrorCodes(d.Connection) {
-			return false
-		}
-
 		gcpConfig := GetConfig(d.Connection)
+
 		if gerr, ok := err.(*googleapi.Error); ok {
+			for _, pattern := range gcpConfig.IgnoreErrorMessages {
+				re := regexp.MustCompile(pattern)
+				result := re.FindString(types.ToString(gerr.Message))
```

Comment:
Does GCP ever send messages that don't have the `Message` property? If so, does this code still handle everything ok?

Reply:
I haven't encountered any errors without a message property. Additionally, we are converting errors of type `googleapi.Error`, which always includes a message property. I've tested it multiple times with various connection combinations, and it's working correctly, handling errors as expected.

If an error does not contain a `message` property, the code block above will not function as it is designed to handle errors based on their error message. In cases where the `message` property is not available, an alternative approach is to configure the `ignore_error_codes` in the .spc file. This will allow the following line of code to handle the error.

According to the [GCP API design documentation](https://cloud.google.com/apis/design/errors), errors include a message property. However, it's important to note that `Error messages are not part of the API surface. They are subject to changes without notice. Application code should not rely heavily on error messages as they may change.`